packages/graph/hash_graph/lib/graph/src/store/postgres/query/property_type.rs

```diff
@@ -0,0 +1,132 @@
+use std::iter::once;
+
+use postgres_types::ToSql;
+use type_system::PropertyType;
+
+use crate::{
+    ontology::PropertyTypeQueryPath,
+    store::postgres::query::{ColumnAccess, Field, Path, PostgresQueryRecord, Table, TableName},
+};
+
+impl<'q> PostgresQueryRecord<'q> for PropertyType {
+    type Field = PropertyTypeField;
+
+    fn base_table() -> Table {
+        Table {
+            name: TableName::PropertyTypes,
+            alias: None,
+        }
+    }
+
+    fn default_fields() -> &'q [Self::Field] {
+        &[PropertyTypeField::Schema, PropertyTypeField::OwnedById]
+    }
+}
+
+/// A [`Field`] available in [`PropertyType`]s.
+///
+/// [`PropertyType`]: type_system::PropertyType
+#[derive(Debug, PartialEq, Eq)]
+pub enum PropertyTypeField {
```

Comment:
Can you help me understand how this is different from the `QueryPath`? They seem very similar, although one obviously has recursive info

Reply:
The path is used inside of structural queries to find a specific column. Fields are used in the `SelectCompiler` to populate the `SELECT <this part> FROM`, this is the reason why `VersionId` and `Schema` are available in the `Field` variants, but not in `Path`.